host.py

```diff
@@ -122,7 +124,38 @@ def _read_output(self, cmd: str, log_level: int = logging.DEBUG) -> Result:
 
         return Result(out, err, exit_code)
 
+    def remove(self, source):
+        if self._hostname == "localhost":
+            if os.path.exists(source):
+                os.remove(source)
+        else:
+            try:
+                ssh_sftp = self._host.open_sftp()
+                ssh_sftp.remove(source)
+            except FileNotFoundError:
+                pass
+
+    def scp(self, source, destination):
+        assert self._hostname != "localhost"
+        ssh_scp = SCPClient(self._host.get_transport())
```

Comment:
paramiko has sftp, we would benefit from having something similar to what we have in one version of this same file in another project (which we should unify at some point).

This is the code, can you be inspired a bit by it?
```
    def put_sftp(self, local_file, remote_file):
        while True:
            try:
                sftp = self._host.open_sftp()
                if not os.path.exists(local_file):
                    print(f"Can't tranfer missing local file {local_file}")
                    sys.exit(-1)
                return sftp.put(local_file, remote_file)
            except Exception as e:
                print(e)
                print(f"Connection lost while trying to open sftpd connection, reconnecting...")
                self.ssh_connect_looped(self._username)
```